src/main/java/com/github/nylle/javafixture/InstanceFactory.java

```diff
@@ -57,27 +57,27 @@ public InstanceFactory(SpecimenFactory specimenFactory) {
         this.random = new PseudoRandom();
     }
 
-    public <T> T construct(final SpecimenType<T> type, CustomizationContext customizationContext) {
+    public <T> T construct(SpecimenType<T> type, CustomizationContext customizationContext) {
         return random.shuffled(type.getDeclaredConstructors())
                 .stream()
                 .filter(x -> Modifier.isPublic(x.getModifiers()))
                 .findFirst()
                 .map(x -> construct(type, x, customizationContext))
-                .orElseGet(() -> manufacture(type, customizationContext));
+                .orElseGet(() -> manufacture(type, customizationContext, new SpecimenException("No public constructor found")));
```

Comment:
We should add the name of the class for which no public constructor was found. It could be that this is not the class I am fixturing but one of its fields.

Reply:
Good point!

Reply:
And if we add something like "trying factory methods next" to the message, it would improve the understandability of the stack trace.

Reply:
Maybe we should not do exceptions at all and implement proper strategies?

I mean, this can certainly be improved. But do you think this is mergeable as a first attempt with improving it on further iterations?

Reply:
Long comment short: Yes, I think we should try it.
Let's label it as experimental because it will only affect diagnosis, not behaviour.